base/src/com/google/idea/blaze/base/sync/projectstructure/OrphanedModuleValidator.java

```diff
@@ -0,0 +1,14 @@
+package com.google.idea.blaze.base.sync.projectstructure;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.module.Module;
+
+/**
+ * Finds if the module is an orphan module, to be not shown in project tree view.
+ */
+public interface OrphanedModuleValidator {
```

Comment:
I'm not sure about name... it is not `validator` but it is up to @tpasternak to decide what would be a good name here

Reply:
updated to ExternalModuleProvider.